src/token/token.js

```diff
@@ -40,18 +41,36 @@ class Token {
 		 * `create` method creates token with initialized value from url.
 		 *
 		 * @name value
-		 * @type {String}
+		 * @member {String} #value
 		 * @observable
 		 * @readonly
-		 * @memberOf Token#
 		 */
 		this.set( 'value', options.initValue );
 
 		/**
+		 * Refresh token function.
+		 *
+		 * @member {Function} #_refreshToken
+		 * @protected
+		 */
+		if ( typeof tokenUrlOrRefreshToken === 'function' ) {
+			this._refreshToken = () => {
+				return tokenUrlOrRefreshToken()
+					.then( value => this.set( 'value', value ) )
+					.then( () => this );
+			};
+		} else {
+			this._refreshToken = this._defaultRefreshToken.bind( this );
+		}
+
+		/**
+		 * A token Url, which is requested by the {@link #_defaultRefreshToken} function.
+		 * An empty string when the callback is provided in the constructor.
+		 *
 		 * @type {String}
 		 * @private
 		 */
-		this._tokenUrl = tokenUrl;
+		this._tokenUrl = typeof tokenUrlOrRefreshToken === 'string' ? tokenUrlOrRefreshToken : '';
```

Comment:
Since this property is used only by `_defaultRefreshToken` I would keep it simple, remove this class property and pass this value only to the default listener.